e2e2/test/images/bert-training/train.py

```diff
@@ -0,0 +1,100 @@
+import os
+import time
+import torch
+import torch.distributed as dist
+from torch.nn.parallel import DistributedDataParallel as DDP
+from transformers import BertForPreTraining, BertTokenizer
+from torch.utils.data import DataLoader, TensorDataset
+import numpy as np
+
+def create_dummy_data(tokenizer, num_samples=100, max_length=128):
+    # Create dummy input data
+    sentences = ["This is a dummy sentence number {}".format(i) for i in range(num_samples)]
+    tokenized_inputs = tokenizer(sentences, max_length=max_length, padding="max_length", truncation=True, return_tensors="pt")
+    labels = tokenized_inputs.input_ids.detach().clone()
+
+    # MLM task: randomly mask some tokens
+    mlm_probability = 0.15
+    input_ids, labels = mask_tokens(tokenized_inputs.input_ids, tokenizer, mlm_probability)
+
+    # NSP task: create dummy pairs
+    next_sentence_labels = torch.randint(0, 2, (num_samples,))
+
+    return TensorDataset(input_ids, tokenized_inputs.attention_mask, labels, next_sentence_labels)
+
+def mask_tokens(inputs, tokenizer, mlm_probability):
+    labels = inputs.clone()
+    probability_matrix = torch.full(labels.shape, mlm_probability)
+    special_tokens_mask = [
+        tokenizer.get_special_tokens_mask(val, already_has_special_tokens=True) for val in labels.tolist()
+    ]
+    probability_matrix.masked_fill_(torch.tensor(special_tokens_mask, dtype=torch.bool), value=0.0)
+    masked_indices = torch.bernoulli(probability_matrix).bool()
+    labels[~masked_indices] = -100  # We only compute loss on masked tokens
+
+    inputs[masked_indices] = tokenizer.convert_tokens_to_ids(tokenizer.mask_token)
+
+    return inputs, labels
+
+def setup(rank, world_size):
+    os.environ['MASTER_ADDR'] = os.environ['MASTER_ADDR']  # Kubernetes sets this
+    os.environ['MASTER_PORT'] = os.environ['MASTER_PORT']  # Kubernetes sets this
```

Comment:
move this to docker `ENV` with defaults

Reply:
What are your thoughts on just setting the defaults here in the script (I should have done this in the first place) versus in the dockerfile?

Reply:
The default values should only be used during local dev, which is why I could see them belonging in the training script. Kubernetes should set them at runtime.

Reply:
> Kubernetes should set them at runtime.

meaning its part of the container spec right? those env's should override what's in the docker `ENV`, so i think it just generally makes sense to put it in the Dockerfile/image definition

Reply:
Yep, it would be in the container spec. Works for me, I'll make the change.

Reply:
@ndbaker1 Done